core/src/com/mygdx/game/models/Round.java

```diff
@@ -0,0 +1,114 @@
+ * maxSelectedBrains: the max nr of selected brains in the eliminationphase
+ * inEliminationPhase: A boolean to tell which round is going
+ *
+ * */
+
+public class Round {
+
+    private ArrayList<Player> players;
+    private HashMap<Player, BrainstormingPhase> brainstormingPhases = new HashMap<>();
+    private HashMap<Player, EliminationPhase> eliminationPhases = new HashMap<>();
+    private HashMap<Player, ArrayList<Brain>> playersBrains = new HashMap<>();
+    private HashMap<Player, Integer> currentBrainNumbers = new HashMap<>();
+    private int maxSelectedBrains;
+    private boolean inEliminationPhase = false;
+
+
+    /**
+     * A constructor that initializes the brainstormingphases for each player.
+     * @param players: List of current players
+     * @param brains: List of brains used this round, if round one these brains should contain no ideas
+     * @param maxHitPoints: The max HP for the wall
+     * @param BRAIN_DAMAGE: The amound of damage the brain does
+     * @param maxSelectedBrains: Max selected brains for the eliminationphase
+     * */
+    public Round(ArrayList<Player> players, ArrayList<Brain> brains, int maxHitPoints, int BRAIN_DAMAGE, int maxSelectedBrains){
+        this.players = players;
+        for(Player player:players){
+            brainstormingPhases.put(player, new BrainstormingPhase(player, maxHitPoints, BRAIN_DAMAGE));
+            playersBrains.put(player, brains);
+            currentBrainNumbers.put(player, 0);
+        }
+
+        this.maxSelectedBrains = maxSelectedBrains;
+    }
+
+    public ArrayList<Player> getPlayers() {
+        return players;
+    }
+
+    public boolean isInEliminationPhase() {
+        return inEliminationPhase;
+    }
+
+    /**
+     * Checks which players still has their walls standing, and returns a list of them
+     * @return A list of players who still have standing walls
+     * */
+    public ArrayList<Player> playersLeft(){
+        ArrayList<Player> remainingPlayers = new ArrayList<>();
+        for (Player player : players){
+            if (brainstormingPhases.get(player).getWall().isStanding()){
+                remainingPlayers.add(player);
+            }
+        }
+        return remainingPlayers;
+    }
+
+    /**
+     * Gets the current brain, and adds the idea on the current brain in the
+     * brainstormingphase connected to the user.
+     * @param player: current player
+     * @param idea: Idea thats going on the brain
+     * @return true if the wall has fallen down, false if not.
+     * */
+    public boolean addBrainInBrainstormingPhase(Player player, String idea){
+        int currentBrainNr = currentBrainNumbers.get(player);
+        Brain brain = playersBrains.get(player).get(currentBrainNr);
+        currentBrainNumbers.put(player, currentBrainNr + 1);
+        BrainstormingPhase brainstormingPhase = brainstormingPhases.get(player);
+        return brainstormingPhase.putIdeaOnBrainAndFire(brain, idea);
+    }
```

Comment:
Personally I would do most of this in the Brainstorming class, and let the controller call that method instead. But we can wait with the decision on what to do until we're surer about how to do things.

Reply:
Most of it is done in the brainstormingclass. The reason it looks complicated is because of the hashmaps and how im retriving the information. There is probably simpler ways of doing this with lambda expressions.

Reply:
```suggestion
    /**
     * Gets the current brain, and adds the idea on the current brain in the
     * brainstormingphase connected to the user.
     * @param player: current player
     * @param idea: Idea thats going on the brain
     * @return true if the wall has fallen down, false if not.
     * */
    public boolean addBrainInBrainstormingPhase(Player player, String idea){
        int currentBrainNr = currentBrainNumbers.get(player);
        currentBrainNumbers.put(player, currentBrainNr + 1);
        return brainstormingPhases.get(player).putIdeaOnBrainAndFire(playersBrains.get(player).get(currentBrainNr), idea);
    }
```


This is another way of doing it, but I feel like the former version is more clean and easier to read. What do you think?

Reply:
I meant that the firing of brains didn't happen in this class at all, but again that might not be the best solution. I agree that the way you wrote it originally is more readable than the suggestion.